fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/provider/foundationdb/FDBRecordContext.java

```diff
@@ -1397,4 +1427,31 @@ public RecordLayerPropertyStorage getPropertyStorage() {
         return config.getPropertyStorage();
     }
 
+    /**
+     * Get key ranges that encountered conflicts when trying to {@link #commit} failed due to {@code NOT_COMMITTED}.
+     * @return list of conflict ranges
+     */
+    @Nullable
+    public List<Range> getNotCommittedConflictingKeys() {
+        return notCommittedConflictingKeys;
+    }
+
+    @SuppressWarnings("PMD.CloseResource")
+    private CompletableFuture<Void> readConflictingKeys() {
+        notCommittedConflictingKeys = new ArrayList<>();
+        final Transaction tr = ensureActive();
+        tr.options().setReadSystemKeys();
```

Comment:
Is this option necessary? I thought the special keyspace (i.e., the `\xff\xff` keyspace) didn't need it, unlike the non-special system keyspace (i.e., the `\xff` keyspace). I could have that wrong, though

Reply:
No, you're right. It isn't necessary.